### PR TITLE
Enable psr6 cache

### PR DESCRIPTION
**Pull request by Tasiobg**
Enable PSR-6 cache with NullAdapter on dev because we don't want to cache locally but we want our code to support it.
Added the dependenci injection into pps.service_factory because "programmes-pages-service" is going to implement caching

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

**Comment by BjornTwachtmann** [comment]
This looks good to me, however I think we should quickly build and load test a branch (can be using apachebench or something simple rather than a full on gattling load test) to ensure there is no crazyness in the new pages-service caching implementation slowing us down before merging.

File: app/config/config.yml [hunk 1]
```diff
@@ -33,6 +33,13 @@ framework:
     # assets: ~
     php_errors:
         log: true
+    cache:
+        pools:
+            cache.programmes:
+                # cache.adapter.psr6 act as a proxy for the adapter we set on the "provider" section,
+                # this is required when and adapter is not supported directly
+                adapter: cache.adapter.psr6
+                provider: cache.null_provider # locally we don't want to use cache
```
Comment (votemike) [inline app/config/config.yml]: Should there be a live/production config as well that overrides this?

Comment (BPScott) [inline app/config/config.yml]: We're not going to setup Redis caching for Clifton just yet - it's currently performant using just the DB and we've still got a bunch of questions we need to answer about what the redis box on AWS shall look like.

We'll add the production config to use Redis once we've finalised the config for the cluster and it's up and running.

**Comment by Tasiobg** [comment]
Should I "Squash and merge" now or when and who does it?

**Pull request merged**